esp/clients/wsdfuaccess/wsdfuaccess.cpp

```diff
@@ -522,20 +528,40 @@ StringBuffer &encodeDFUFileMeta(StringBuffer &metaInfoBlob, IPropertyTree *metaI
      * Should be part of the same configuration setup.
      */
 #ifdef _USE_OPENSSL
-    if (metaInfo->hasProp("keyPairName") && environment) // without it, meta data is not encrypted
+    if (metaInfo->hasProp("keyPairName") && (isContainerized() || environment)) // without it, meta data is not encrypted
     {
         MemoryBuffer metaInfoBlob;
         metaInfo->serialize(metaInfoBlob);
+        const char *keyPairName = metaInfo->queryProp("keyPairName"); // NB: in container mode, this is cert. name
 
-        const char *keyPairName = metaInfo->queryProp("keyPairName");
-        const char *privateKeyFName = environment->getPrivateKeyPath(keyPairName);
+        const char *privateKeyFName = nullptr;
+        Owned<IPropertyTree> metaInfoEnvelope = createPTree();
+#ifdef _CONTAINERIZED
+        /* may revisit, for now encode the public certificate in the request. NB: this is an approached used for JWT token delegation.
```

Comment:
an approach used

Reply:
Just a note: I think the only reason to revisit is if we care about request size.  Shouldn't be security issues sending a public certificate.

Reply:
Will amended the comment.